src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerModifierAnalyzer.cs

```diff
@@ -0,0 +1,140 @@
+					{ "IsContainingTypeSealed",				handler.Symbol.ContainingType.IsSealed.ToString()}
+				}
+				.ToImmutableDictionary();
+
+				if (handler.Symbol.MethodKind == MethodKind.ExplicitInterfaceImplementation)
+				{
+					context.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations, location, properties));
+				}
+				else if (!HelperMethods.ImplementsInterface(handler.Symbol, pxContext))
+				{
+					var targetAccessibility = handler.Symbol.ContainingType.IsSealed ? Accessibility.Public : Accessibility.Protected;
+
+					if (!(handler.Symbol.IsVirtual || handler.Symbol.ContainingType.IsSealed) || handler.Symbol.DeclaredAccessibility != targetAccessibility)
+					{
+						context.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual, location, properties));
+					}
+				}
+			}
+		}
+
+		private static class HelperMethods
+		{
+			internal static bool ImplementsInterface(IMethodSymbol method, PXContext pxContext)
+			{
+				return method
+					.ContainingType
+					.AllInterfaces
+					.SelectMany(i => i.GetMethods(method.Name))
+					.Any(m =>
+						IsEventHandlerInfosMatch(method, m, pxContext) &&
+						IsImplicitInterfaceImplementation(method, m)
+					);
+			}
+
+			private static bool IsEventHandlerInfosMatch(IMethodSymbol method, IMethodSymbol other, PXContext pxContext)
+			{
+				return method.GetEventHandlerInfo(pxContext).Equals(other.GetEventHandlerInfo(pxContext));
+			}
+
+			private static bool IsImplicitInterfaceImplementation(IMethodSymbol method, IMethodSymbol interfaceMethod)
+			{
+				if (method.Name != interfaceMethod.Name)
+				{
+					return false;
+				}
+
+				if (method.Parameters.Length != interfaceMethod.Parameters.Length)
+				{
+					return false;
+				}
+
+				if (!method.ReturnType.Equals(interfaceMethod.ReturnType, SymbolEqualityComparer.Default))
+				{
+					return false;
+				}
+
+				if (method.Arity != interfaceMethod.Arity)
+				{
+					return false;
+				}
+
+				if (method.IsGenericMethod != interfaceMethod.IsGenericMethod)
+				{
+					return false;
+				}
+
+				if (method.IsGenericMethod)
+				{
+					if (method.TypeArguments.Length != interfaceMethod.TypeArguments.Length)
+					{
+						return false;
+					}
+
+					for (int i = 0; i < method.TypeArguments.Length; i++)
+					{
+						if (!method.TypeArguments[i].Equals(interfaceMethod.TypeArguments[i], SymbolEqualityComparer.Default))
+						{
+							return false;
+						}
+					}
+				}
+
+				for (int i = 0; i < method.Parameters.Length; i++)
+				{
+					if (!method.Parameters[i].Type.Equals(interfaceMethod.Parameters[i].Type, SymbolEqualityComparer.Default))
+					{
+						return false;
+					}
+				}
+
+				var implicitImplementation = method.ContainingType.FindImplementationForInterfaceMember(interfaceMethod);
+
+				return implicitImplementation?.Equals(method, SymbolEqualityComparer.Default) ?? false;
```

Comment:
The check that `method`'s overriden methods are equal to the `implicitImplementation` is missing.

Reply:
Can you please explain this a little further? If the method overrides a parent method (if it has "override" modifier), we skip altogether anyway.